### PR TITLE
[MISC] deprecate seqan3::views::convert

### PR DESCRIPTION
**Pull request by eseiler**
Resolves https://github.com/seqan/product_backlog/issues/324

### CODE REVIEW

**Review by MitraDarja — APPROVED** [review]

File: doc/cookbook/alphabet_conversion.cpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    seqan3::debug_stream << view1 << '\n'; // ACGT
+
+    // Convert dna5 to dna4.
+    // General case: Perfect forward.
+    auto view2 = sequence2
+               | std::views::take(8)
+               | std::views::transform([] (auto && in)
+                   {
+                       return static_cast<seqan3::dna4>(std::forward<decltype(in)>(in));
+                   });
```
Comment (marehr) [inline doc/cookbook/alphabet_conversion.cpp:34]:
```suggestion
               | std::views::transform([] (auto && in)
                 {
                     return static_cast<seqan3::dna4>(std::forward<decltype(in)>(in));
                 });
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -191,6 +191,12 @@ regression test suite and patches at https://github.com/seqan/seqan3/tree/master
     * `seqan3::get<seqan3::field::cigar>(record)` => `record.cigar_sequence()`
     * `seqan3::get<seqan3::field::tags>(record)` => `record.tags()`
 
+#### Range
+
+* Deprecated `seqan3::views::convert`. There is no direct replacement, see
+  [SeqAn3 Cookbook - Converting a range of an alphabet](https://docs.seqan.de/seqan/3.0.3/cookbook.html) for an example
+  of using `std::views::transform` for conversion ([\#2524](https://github.com/seqan/seqan3/pull/2524)).
```
Comment (marehr) [inline CHANGELOG.md:198]:
```suggestion
* Me made `seqan3::views::convert` NOAPI and moved it to `seqan3/utility/views/convert.hpp`. You can still use `seqan3::views::convert` in the meantime, but we would encourage using `std::views::transform` instead as shown in [SeqAn3 Cookbook - Converting a range of an alphabet](https://docs.seqan.de/seqan/3.0.3/cookbook.html).
```

Can we directly link to that entry? cookbook isn't small xD

File: doc/cookbook/alphabet_conversion.cpp (hunk 1 above)
Comment (marehr) [reply doc/cookbook/alphabet_conversion.cpp:34]: Thank you! nice snippet!

File: test/snippet/alphabet/composite/semialphabet_any.cpp [hunk 3]
```diff
@@ -33,24 +32,32 @@ void algorithm(std::vector<seqan3::semialphabet_any<10> > & r, bool is_murphy)
...
 // They type erase the different arguments to the same type and encode the type information as a run-time parameter
 void algo_pre(seqan3::aa10li_vector const & v)
 {
-    std::vector<seqan3::semialphabet_any<10> > tmp = v | seqan3::views::convert<seqan3::semialphabet_any<10>>
-                                                       | seqan3::views::to<std::vector>;
+    std::vector<seqan3::semialphabet_any<10> > tmp = v
+                                                   | std::views::transform([] (auto const & in)
+                                                       {
+                                                           return static_cast<seqan3::semialphabet_any<10>>(in);
+                                                       })
```
Comment (marehr) [inline test/snippet/alphabet/composite/semialphabet_any.cpp:48]:
```suggestion
                                                   | std::views::transform([] (auto const & in)
                                                     {
                                                         return static_cast<seqan3::semialphabet_any<10>>(in);
                                                     })
```

File: test/snippet/alphabet/composite/semialphabet_any.cpp [hunk 4]
```diff
@@ -33,24 +32,32 @@ void algorithm(std::vector<seqan3::semialphabet_any<10> > & r, bool is_murphy)
...
 
 void algo_pre(seqan3::aa10murphy_vector const & v)
 {
-    std::vector<seqan3::semialphabet_any<10> > tmp = v | seqan3::views::convert<seqan3::semialphabet_any<10>>
-                                                       | seqan3::views::to<std::vector>;
+    std::vector<seqan3::semialphabet_any<10> > tmp = v
+                                                   | std::views::transform([] (auto const & in)
+                                                       {
+                                                           return static_cast<seqan3::semialphabet_any<10>>(in);
+                                                       })
```
Comment (marehr) [inline test/snippet/alphabet/composite/semialphabet_any.cpp:59]:
```suggestion
                                                   | std::views::transform([] (auto const & in)
                                                     {
                                                         return static_cast<seqan3::semialphabet_any<10>>(in);
                                                     })
```

File: test/snippet/range/views/range_view_to_rank.cpp [hunk 5]
```diff
@@ -5,25 +5,24 @@
...
 
     std::vector<seqan3::dna4q> qcvec{{'C'_dna4, seqan3::phred42{0}}, {'A'_dna4, seqan3::phred42{7}},
                                      {'G'_dna4, seqan3::phred42{5}}, {'T'_dna4, seqan3::phred42{3}},
                                      {'G'_dna4, seqan3::phred42{7}}, {'A'_dna4, seqan3::phred42{4}},
                                      {'C'_dna4, seqan3::phred42{30}}, {'T'_dna4, seqan3::phred42{16}},
                                      {'A'_dna4, seqan3::phred42{23}}};
-    auto v4 = qcvec | seqan3::views::to_rank | seqan3::views::convert<unsigned>;
-    seqan3::debug_stream << v4 << '\n'; // [1,28,22,15,30,16,121,67,92]
+    auto v4 = qcvec | seqan3::views::to_rank;
+    seqan3::debug_stream << v4 << '\n'; // [42,7,89,129,91,4,72,142,23]
```
Comment (marehr) [inline test/snippet/range/views/range_view_to_rank.cpp:27]: ups? this was wrong?

**Review by marehr — COMMENTED** [review]
small things, thank you so much :)

File: test/snippet/range/views/range_view_to_rank.cpp (hunk 5 above)
Comment (eseiler) [reply test/snippet/range/views/range_view_to_rank.cpp:27]: Yup

**Review by marehr — APPROVED** [review]
thank you!!

**Pull request merged**